src/components/ScreenContainer.tsx

```diff
@@ -0,0 +1,31 @@
+import { Platform, View } from 'react-native';
+import React from 'react';
+import { ScreenContainerProps } from 'react-native-screens';
+import { isNativePlatformSupported, screensEnabled } from '../core';
+
+// Native components
+import ScreenContainerNativeComponent from '../fabric/ScreenContainerNativeComponent';
+import ScreenNavigationContainerNativeComponent from '../fabric/ScreenNavigationContainerNativeComponent';
+
+export const NativeScreenContainer: React.ComponentType<ScreenContainerProps> =
+  ScreenContainerNativeComponent as any;
```

Comment:
Is there no way that there would be no `any` here?

Reply:
I would love to not have any `any` here, but this would require us to type all of the native components (that are default exports with codegenNativeComponent - I don't even know if we could type those codegens 🤔)

Reply:
Aren't they automatically typed with the props you add as their `NativeProps` type to `codegenNativeComponent`?

Reply:
I believe they are not (or am I doing something wrong?). Earlier, when there were optional requires in the code typescript didn't care about the types (since it didn't know what we are importing). Right now it can't handle proper typing (and I see some reasons there - you've mentioned that this component should be automatically typed and maybe you're right, but some NativeProps differs between what types we wanna provide for the users (here - ScreenContainerProps) and what we want to provide for Codegen (e.g. BubblingEventHandlers). You can find such case in Search Bar, where for the search bar commands we provide SearchBarCommandsType for the user and we expect that this native component will match syntactically whereas for the codegen we want to expect that `focus` event will be BubblingEventHandler.

<img width="1063" alt="image" src="https://github.com/software-mansion/react-native-screens/assets/23281839/8bfb1a66-51b8-419a-8e1e-26d51d6354d5">

Reply:
You're probably right, but we still don't provide exact same types as we provide in NativeProps. Earlier, when there were optional requires in the code, Typescript didn't care about the types (since it didn't know what we are importing). Right now it can't handle proper typing (and I see some reasons there - you've mentioned that this component should be automatically typed and maybe you're right, but some NativeProps differs between what types we wanna provide for the users (here - ScreenContainerProps) and what we want to provide for Codegen (e.g. BubblingEventHandlers). You can find such case in Search Bar, where for the search bar commands we provide SearchBarCommandsType for the user and we expect that this native component will match syntactically whereas for the codegen we want to expect that `focus` event will be BubblingEventHandler.

<img width="1063" alt="image" src="https://github.com/software-mansion/react-native-screens/assets/23281839/8bfb1a66-51b8-419a-8e1e-26d51d6354d5">